web-client/src/applicationContext.ts

```diff
@@ -365,6 +365,39 @@ const reduce = ImageBlobReduce({
 
 let user;
 let broadcastChannel;
+const clientSupportsES2022 = (() => {
```

Comment:
Most sources I saw recommended doing feature detection rather than looking at agent strings, etc.

Reply:
Huh interesting...how do you decide what features to check for? Did you "just know", is there a known/recommended list somewhere, something else?

Reply:
Great question! The most thorough thing to do is to go through the library and see exactly which features the library needs. This is overkill and also assumes you can keep up-to-date.

The next best thing is to find some minimal set of ES2022 features: if a browser client fails to implement any of them, you know that the client does not conform to ES2022.

The features checked here are therefore based on the following:
- The updated comment in the sneakily compatability-breaking release: https://github.com/mozilla/pdfjs-dist/compare/v2.12.313...v2.13.216.
- The message logs, which indicate issues with private fields and with Array prototype `at`
- Basic EMCAScript updates, like `hasOwn` instead of `hasOwnProperty`

Is this 100% foolproof? No: a random browser client could choose to implement some but not all ES2022 features. But this should be rare, and I think it is better to load the modern library when possible rather than always only loading the legacy build.

Reply:
"See the ticket for screenshots" lol didn't link the ticket. Here we go: https://github.com/flexion/ef-cms/issues/10531

Reply:
Thanks for the explanation! That makes a lot of sense. I wonder if it's worth somehow documenting the rationale for what we're checking next to where we're checking it? I'm personally unsure, and regardless, don't think that's worth blocking PR approval.